### PR TITLE
update Samtools, bcftools, and htslib versions

### PR DESCRIPTION
**Pull request by downloadico**

### CODE REVIEW

File: var/spack/repos/builtin/packages/bcftools/package.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
```
Comment (kwryankrattiger) [inline var/spack/repos/builtin/packages/bcftools/package.py:1]: Fixup the copyright headers

**Comment by kwryankrattiger** [comment]
LGTM after the copyright headers are reverted to the 2023.

File: var/spack/repos/builtin/packages/bcftools/package.py (hunk 1 above)
Comment (downloadico) [reply var/spack/repos/builtin/packages/bcftools/package.py:1]: OOPS!  We *did* increment the year recently, didn't we?! ;)

**Comment by tgamblin** [comment]
We can merge this and fix that later.

File: var/spack/repos/builtin/packages/samtools/package.py [hunk 2]
```diff
@@ -41,6 +41,7 @@ class Samtools(Package):
     depends_on("python", type="run")
 
     # htslib became standalone @1.3.1, must use corresponding version
+    depends_on("htslib@1.16", when="@1.16.1")
```
Comment (kwryankrattiger) [inline var/spack/repos/builtin/packages/samtools/package.py:44]: @downloadico It looks like you added the dependency here, but not the new version :)

**Pull request merged**